btb/tuning/hyperparams/categorical.py

```diff
@@ -37,7 +37,10 @@ def __init__(self, choices, default=None):
         Creates an instance with a list of ``choices`` and fit an instance of
         ``sklearn.preprocessing.OneHotEncoder`` with those values.
         """
-        self.default = default or choices[0]
+        if default not in choices:
```

Comment:
This will silently overwrite an invalid default:

```
>>> chp = CategoricalHyperParam(choices=['a', 'b', 'c'], default='d')
>>> chp.default
'a'
```

Instead of this, and since `None` could also a valid option, I would set something dummy but unique as default and raise an exception if something other than that is given which is not within choices.

```
class CategoricalHyperParam(...):
    NO_DEFAULT = object()
    ...
    def __init__(self, choices, default=NO_DEFAULT):
        if default is self.NO_DEFAULT:
            self.default = choices[0]
        elif default not in choices:
            raise ValueError('`default` not within `choices`')
        else:
            self.default = default
```